package/MDAnalysis/analysis/pca.py

```diff
@@ -0,0 +1,234 @@
+        trajectory.
+    transform(traj=None, n_components=None)
+        Take the original trajectory and project it onto the principal
+        components.
+    inverse_tranform(pca_space)
+        Take a pca_space and map it back onto the trajectory used to create it.
+    """
+
+    def __init__(self, atomgroup, select='all', demean=True,
+                 n_components=None, **kwargs):
+        """
+        Parameters
+        ----------
+        atomgroup: MDAnalysis atomgroup
+            AtomGroup to be used for PCA.
+        select: string, optional
+            A valid selection statement for choosing a subset of atoms from
+            the atomgroup.
+        demean: boolean, optional
+            If False, the trajectory will not be aligned to a reference
+            and there will be no demeaning of covariance data.
+        n_components : int, optional
+            The number of principal components to be saved, default saves
+            all principal components, Default: -1
+        start : int, optional
+            First frame of trajectory to use for generation
+            of covariance matrix, Default: None
+        stop : int, optional
+            Last frame of trajectory to use for generation
+            of covariance matrix, Default: None
+        step : int, optional
+            Step between frames of trajectory to use for generation
+            of covariance matrix, Default: None
+        """
+        super(PCA, self).__init__(atomgroup.universe.trajectory,
+                                  **kwargs)
+        self._u = atomgroup.universe
+        # for transform function
+        self._demean = demean
+        # access 0th index
+        self._u.trajectory[0]
+        # reference will be 0th index
+        self._reference = self._u.atoms.select_atoms(select)
+        self._atoms = atomgroup.select_atoms(select)
+        self.n_components = n_components
+        self._n_atoms = self._atoms.n_atoms
+        self._calculated = False
+
+    def _prepare(self):
+        n_dim = self._n_atoms * 3
+        self.cov = np.zeros((n_dim, n_dim))
+        if self._demean:
+            self._ref_atoms = self._reference
+            self._ref_cog = self._ref_atoms.center_of_geometry()
+
+    def _single_frame(self):
+        if self._demean:
+            # TODO make this more functional, initial proof of concept
+            mobile_atoms, old_rmsd = _fit_to(self._atoms.positions,
+                                             self._ref_atoms.positions,
+                                             self._atoms, self._ref_cog,
+                                             self._atoms.center_of_geometry())
+            # now all structures are aligned to reference
+            x = mobile_atoms.positions.ravel()
+        else:
+            x = self._atoms.positions.ravel()
+        self.cov += np.dot(x[:, np.newaxis], x[:, np.newaxis].T)
+
+    def _conclude(self):
+        self.cov /= self.n_frames - 1
```

Comment:
OK looks like [gromacs covar](http://repo.or.cz/gromacs.git/blob/HEAD:/src/gromacs/gmxana/gmx_covar.cpp) is the better tool to benchmark your PCA against. It also does calculate an average structure (your mean) but doesn't use it in the standard configuration and only does a least-square fit.

I'll check tomorrow how to extract the projections from the `gmx anaeig` to get comparison values.